libenkf/src/rng_config.c

```diff
@@ -110,11 +140,24 @@ rng_manager_type * rng_config_alloc_rng_manager( const rng_config_type * rng_con
   const char * seed_load  = rng_config_get_seed_load_file( rng_config );
   rng_manager_type * rng_manager;
 
-  if (seed_load && util_file_exists( seed_load ))
-    rng_manager = rng_manager_alloc( seed_load );
+  if (rng_config->random_seed) {
+    char * formatted_seed = rng_config_alloc_formatted_random_seed(rng_config);
+    rng_manager = rng_manager_alloc(formatted_seed);
+  }
+  else if (seed_load && util_file_exists( seed_load ))
+    rng_manager = rng_manager_alloc_load( seed_load );
   else
     rng_manager = rng_manager_alloc_random( );
 
+  res_log_add_fmt_message(LOG_CRITICAL, stdout, "To repeat this experiment, add the following random seed to your config file:");
+  if (rng_config->random_seed) {
+    res_log_add_fmt_message(LOG_CRITICAL, stdout, "RANDOM_SEED %s", rng_config->random_seed);
+  } else {
+    unsigned int random_seed [4];
```

Comment:
The lines 152-159 could be replaced with a call: `rng_config_log_seed( rng_config )`- which again unconditionally called the internal rng to get state; should not be necessary to distinguish between seeds explicitly provided by the user and default initialized seeds?? If this is indeed possible we can also remove the `rng_manager_get_state()` function completely.

Reply:
The thing is that the journey from a random seed in the configuration file to the internal state of the `rng_manager` is not injective. Due to this, if specifying `RANDOM_SEED 1234` in the configuration file, you might very well be asked to add `123412341234123412341234123412341234123` to the configuration file to repeat the experiment, which is kind of odd and unexpected.

Reply:
>Due to this, if specifying RANDOM_SEED 1234 in the configuration file, you might very well be asked to add 123412341234123412341234123412341234123 to the configuration file to repeat the experiment, which is kind of odd and unexpected.

Well - we can live with that - in my opinon.